cmd/hostpaths/hostpaths.go

```diff
@@ -420,6 +407,75 @@ func mapHostPaths(ctx context.Context, pManager, vManager manager.Manager) {
 	}, time.Second*5)
 }
 
+func getPhysicalPodMap(ctx context.Context, options *context2.VirtualClusterOptions, pManager manager.Manager) (PhysicalPodMap, error) {
```

Comment:
I like the flow much better now, although the function is a little bit long imho. Could you write some unit tests for it? 🙏 (you will have to separate the first part where you fetch the pods and the second part where you do the filtering, tests would only be necessary for the second part)

Reply:
I have extracted the filtering logic in its own function and added unit tests for it and ensured coverage. Right after the filtering part we have a loop that builds the PodMap based on the presence of the actual paths on the host. Unfortunately it is not test friendly in its current form. We can refactor later and implement a fake mapper and then mock such functions so that even they can be covered in tests. We have done something similar for the central-hostpath-mapper. It can be taken up as a separate exercise for this repository later.

Reply:
yeah it's kinda true, I may have underestimated a little bit the amount of changes required to do so, and that would blow the scope of the pr out of proportion.

Reply:
There is already one, though its more related to e2e and helm unit tests. Will check if I can add this to it else will create a separate one.